Convert WebKit WebExtension ARC code to unified source compiling

darinadler opened the pull request:
#### 729d64bab05f34595cdb89f97889b354fdec8328
<pre>
Convert WebKit WebExtension ARC code to unified source compiling
<a href="https://rdar.apple.com/160949824">rdar://160949824</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=299190">https://bugs.webkit.org/show_bug.cgi?id=299190</a>

Reviewed by Timothy Hatcher and Keith Miller.

Added support to make separate unified source files for ARC and non-ARC
Objective-C++ code. Used it in WebKit for extensions ARC code.

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
Renamed unified source files to use -nonARC.mm suffix.

* Source/JavaScriptCore/Scripts/generate-unified-sources.sh:
Update for new non-ARC category.

* Source/JavaScriptCore/SourcesCocoa.txt: Added @nonARC to all the
non-ARC Objective-C++ source files.

* Source/JavaScriptCore/UnifiedSources-output.xcfilelist:
Renamed unified source files to use -nonARC.mm suffix.

* Source/JavaScriptCore/inspector/remote/SourcesCocoa.txt: Added
@nonARC to all the non-ARC Objective-C++ source files.

* Source/WTF/Scripts/generate-unified-source-bundles.rb: Added
support for separate ARC and non-ARC Objective-C++ sources.

* Source/WebCore/Scripts/generate-unified-sources.sh:
Update for new non-ARC category.

* Source/WebCore/SourcesCocoa.txt: Added @nonARC to all the
non-ARC Objective-C++ source files.

* Source/WebCore/UnifiedSources-output.xcfilelist:
Renamed unified source files to use -nonARC.mm suffix.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
Renamed unified source files to use -nonARC.mm suffix.

* Source/WebKit/Platform/SourcesCocoa.txt: Added @nonARC to all the
non-ARC Objective-C++ source files.

* Source/WebKit/Scripts/generate-unified-sources.sh:
Update for new non-ARC category.

* Source/WebKit/SourcesCocoa.txt: Added @nonARC to all the
non-ARC Objective-C++ source files. Added the WebExtensions
source files.

* Source/WebKit/UnifiedSources-output.xcfilelist:
Renamed unified source files to use -nonARC.mm suffix.
Added the new files with the -ARC.mm suffix.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
Renamed unified source files to use -nonARC.mm suffix.
Added new unified source files with the -ARC.mm suffix.
Turned off compilation of all the individual WebExtension
source file that are now compiled in unified sources.

* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPICommandsCocoa.mm:
(WebKit::toAPI):
(WebKit::WebExtensionContextProxy::dispatchCommandsChangedEvent):
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPICookiesCocoa.mm:
(WebKit::toWebAPI):
(WebKit::WebExtensionAPICookies::parseCookieDetails):
(WebKit::WebExtensionAPICookies::get):
(WebKit::WebExtensionAPICookies::set):
(WebKit::WebExtensionAPICookies::remove):
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIMenusCocoa.mm:
(WebKit::WebExtensionAPIMenus::parseCreateAndUpdateProperties):
(WebKit::WebExtensionContextProxy::dispatchMenusClickedEvent):
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIScriptingCocoa.mm:
(WebKit::toWebAPI):
(WebKit::WebExtensionAPIScripting::parseCSSInjectionOptions):
(WebKit::WebExtensionAPIScripting::parseRegisteredContentScripts):
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIWebRequestCocoa.mm:
(WebKit::webRequestDetailsForResourceLoad):
(WebKit::convertHeaderFieldsToWebExtensionFormat):
(WebKit::WebExtensionContextProxy::resourceLoadDidCompleteWithError):
* Source/WebKit/WebProcess/Extensions/Cocoa/_WKWebExtensionWebNavigationURLFilter.mm:
(-[_WKWebExtensionWebNavigationURLFilter initWithDictionary:outErrorMessage:]):
* Source/WebKit/WebProcess/Extensions/Cocoa/_WKWebExtensionWebRequestFilter.mm:
(-[_WKWebExtensionWebRequestFilter initWithDictionary:outErrorMessage:]):
(-[_WKWebExtensionWebRequestFilter _initializeWithDictionary:]):
Use @&quot;&quot; directly instead of named @&quot;&quot; constants to sidestep issue with
duplicate versions of the same string constant.

* Source/WebKitLegacy/SourcesCocoa.txt: Added @nonARC to all the
non-ARC Objective-C++ source files.

* Source/WebKitLegacy/UnifiedSources-output.xcfilelist:
Renamed unified source files to use -nonARC.mm suffix.

* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
Renamed unified source files to use -nonARC.mm suffix.

* Source/WebKitLegacy/scripts/generate-unified-sources.sh:
Update for new non-ARC category.

* Tools/TestWebKitAPI/Scripts/generate-unified-sources.sh:
Update for new non-ARC category.

* Tools/TestWebKitAPI/SourcesCocoa.txt: Added @nonARC to all the
non-ARC Objective-C++ source files.

* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
Renamed unified source files to use -nonARC.mm suffix.

* Tools/TestWebKitAPI/UnifiedSources-output.xcfilelist:
Renamed unified source files to use -nonARC.mm suffix.

Canonical link: <a href="https://commits.webkit.org/300647@main">https://commits.webkit.org/300647@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aa99c5a4d702a052bd0b44eb4480e4d9d3509854

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123326 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43041 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33737 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/130037 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75442 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/18e6db58-b078-4779-b324-3d605c6095b9) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/125203 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43764 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51635 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93744 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62192 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/92128312-0c87-41b7-abd1-bc26fbe2ec99) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126279 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34872 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110343 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74374 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b7b97899-3628-4f37-8e80-8699a6d87781) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/122682 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33838 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28502 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73555 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/115485 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104586 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28728 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132754 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/121857 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50276 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38257 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102237 "2 flakes 53 failures") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50652 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106567 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102090 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25961 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47442 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25665 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/47062 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50131 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/55892 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/152212 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49602 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/38901 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/52952 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51280 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

kmiller68 reviewed (approved):
Nice! r=me too.

> Use @"" directly instead of named @"" constants to sidestep issue with
duplicate versions of the same string constant.

Is this something that would matter for performance / memory? I don't know enough about Obj-C to know why we have these forward defined to begin with. Should we have a `StaticNSStrings.h`?

darinadler committed:
Convert WebKit WebExtension ARC code to unified source compiling
rdar://160949824
https://bugs.webkit.org/show_bug.cgi?id=299190

Reviewed by Timothy Hatcher and Keith Miller.

Added support to make separate unified source files for ARC and non-ARC
Objective-C++ code. Used it in WebKit for extensions ARC code.

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
Renamed unified source files to use -nonARC.mm suffix.

* Source/JavaScriptCore/Scripts/generate-unified-sources.sh:
Update for new non-ARC category.

* Source/JavaScriptCore/SourcesCocoa.txt: Added @nonARC to all the
non-ARC Objective-C++ source files.

* Source/JavaScriptCore/UnifiedSources-output.xcfilelist:
Renamed unified source files to use -nonARC.mm suffix.

* Source/JavaScriptCore/inspector/remote/SourcesCocoa.txt: Added
@nonARC to all the non-ARC Objective-C++ source files.

* Source/WTF/Scripts/generate-unified-source-bundles.rb: Added
support for separate ARC and non-ARC Objective-C++ sources.

* Source/WebCore/Scripts/generate-unified-sources.sh:
Update for new non-ARC category.

* Source/WebCore/SourcesCocoa.txt: Added @nonARC to all the
non-ARC Objective-C++ source files.

* Source/WebCore/UnifiedSources-output.xcfilelist:
Renamed unified source files to use -nonARC.mm suffix.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
Renamed unified source files to use -nonARC.mm suffix.

* Source/WebKit/Platform/SourcesCocoa.txt: Added @nonARC to all the
non-ARC Objective-C++ source files.

* Source/WebKit/Scripts/generate-unified-sources.sh:
Update for new non-ARC category.

* Source/WebKit/SourcesCocoa.txt: Added @nonARC to all the
non-ARC Objective-C++ source files. Added the WebExtensions
source files.

* Source/WebKit/UnifiedSources-output.xcfilelist:
Renamed unified source files to use -nonARC.mm suffix.
Added the new files with the -ARC.mm suffix.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
Renamed unified source files to use -nonARC.mm suffix.
Added new unified source files with the -ARC.mm suffix.
Turned off compilation of all the individual WebExtension
source file that are now compiled in unified sources.

* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPICommandsCocoa.mm:
(WebKit::toAPI):
(WebKit::WebExtensionContextProxy::dispatchCommandsChangedEvent):
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPICookiesCocoa.mm:
(WebKit::toWebAPI):
(WebKit::WebExtensionAPICookies::parseCookieDetails):
(WebKit::WebExtensionAPICookies::get):
(WebKit::WebExtensionAPICookies::set):
(WebKit::WebExtensionAPICookies::remove):
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIMenusCocoa.mm:
(WebKit::WebExtensionAPIMenus::parseCreateAndUpdateProperties):
(WebKit::WebExtensionContextProxy::dispatchMenusClickedEvent):
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIScriptingCocoa.mm:
(WebKit::toWebAPI):
(WebKit::WebExtensionAPIScripting::parseCSSInjectionOptions):
(WebKit::WebExtensionAPIScripting::parseRegisteredContentScripts):
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIWebRequestCocoa.mm:
(WebKit::webRequestDetailsForResourceLoad):
(WebKit::convertHeaderFieldsToWebExtensionFormat):
(WebKit::WebExtensionContextProxy::resourceLoadDidCompleteWithError):
* Source/WebKit/WebProcess/Extensions/Cocoa/_WKWebExtensionWebNavigationURLFilter.mm:
(-[_WKWebExtensionWebNavigationURLFilter initWithDictionary:outErrorMessage:]):
* Source/WebKit/WebProcess/Extensions/Cocoa/_WKWebExtensionWebRequestFilter.mm:
(-[_WKWebExtensionWebRequestFilter initWithDictionary:outErrorMessage:]):
(-[_WKWebExtensionWebRequestFilter _initializeWithDictionary:]):
Use @"" directly instead of named @"" constants to sidestep issue with
duplicate versions of the same string constant.

* Source/WebKitLegacy/SourcesCocoa.txt: Added @nonARC to all the
non-ARC Objective-C++ source files.

* Source/WebKitLegacy/UnifiedSources-output.xcfilelist:
Renamed unified source files to use -nonARC.mm suffix.

* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
Renamed unified source files to use -nonARC.mm suffix.

* Source/WebKitLegacy/scripts/generate-unified-sources.sh:
Update for new non-ARC category.

* Tools/TestWebKitAPI/Scripts/generate-unified-sources.sh:
Update for new non-ARC category.

* Tools/TestWebKitAPI/SourcesCocoa.txt: Added @nonARC to all the
non-ARC Objective-C++ source files.

* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
Renamed unified source files to use -nonARC.mm suffix.

* Tools/TestWebKitAPI/UnifiedSources-output.xcfilelist:
Renamed unified source files to use -nonARC.mm suffix.

Canonical link: https://commits.webkit.org/300647@main
